martenbuild.cs

```diff
@@ -8,46 +8,33 @@ namespace martenbuild
 {
     class MartenBuild
     {
-        private const string RESULTS_DIR = "results";
         private const string BUILD_VERSION = "3.0.0";
 
         static void Main(string[] args)
         {
-            OperatingSystem os = Environment.OSVersion;
+            var platformID = Environment.OSVersion.Platform;
 
-            PlatformID pid = os.Platform;
-
-            var compileTarget = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("config")) 
-                ? "debug" 
-                : Environment.GetEnvironmentVariable("config");
-
-            var connection = Environment.GetEnvironmentVariable("connection");
+            var configuration = Environment.GetEnvironmentVariable("config");
+            configuration = string.IsNullOrEmpty(configuration) ? "debug" : configuration;
 
             Target("ci", DependsOn("connection", "default", "pack"));
-            
+
             Target("default", DependsOn("mocha", "test", "storyteller"));
 
             Target("clean", () =>
             {
-                if (Directory.Exists(RESULTS_DIR))
-                {
-                    Directory.Delete(RESULTS_DIR, true);
-                }
-                
-                if (Directory.Exists("artifacts"))
-                {
-                    Directory.Delete("artifacts", true);
-                }
+                EnsureDirectoryDeleted("results");
+                EnsureDirectoryDeleted("artifacts");
             });
 
             Target("connection", () =>
             {
-                File.WriteAllText("src/Marten.Testing/connection.txt", connection);
+                File.WriteAllText("src/Marten.Testing/connection.txt", Environment.GetEnvironmentVariable("connection"));
             });
 
             Target("install", () =>
             {
-                if (pid == PlatformID.Unix || pid == PlatformID.MacOSX)
+                if (platformID == PlatformID.Unix || platformID == PlatformID.MacOSX)
```

Comment:
Another refactoring area I am observing is checking for the platform, have a helper method as below
```csharp
private static bool IsWindowsPlatform()
{
    var platformID = Environment.OSVersion.Platform;
    
    if (platformID == PlatformID.Unix || platformID == PlatformID.MacOSX)
    {
	return false;
    }

    return true;
}

// change the appropriate code as below to simplify it
IsWindowsPlatform ? Run("cmd.exe", "/c npm run test") : Run("npm", "run test");
```

Reply:
@adamralph I will take care of applying the above.

Reply:
@mysticmind given that the choice between `Run("npm", "...");` and `Run("cmd.exe", "/c npm ...");` is made in two places (`install` and `mocha` targets) I would consider creating a `RunNpm(string args)` method to wrap that decision.